docs/source/en/model_doc/clip.md

```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
+    device_map="auto",
+    attn_implementation="flash_attention_2" # FlashAttention only supports Ampere GPUs or newer
+)
 
-- [Fine tuning CLIP with Remote Sensing (Satellite) images and captions](https://huggingface.co/blog/fine-tune-clip-rsicd), a blog post about how to fine-tune CLIP with [RSICD dataset](https://github.com/201528014227051/RSICD_optimal) and comparison of performance changes due to data augmentation.
-- This [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text) shows how to train a CLIP-like vision-text dual encoder model using a pre-trained vision and text encoder using [COCO dataset](https://cocodataset.org/#home).
+processor = AutoProcessor.from_pretrained("openai/clip-vit-base-patch32")
 
-<PipelineTag pipeline="image-to-text"/>
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+image = Image.open(requests.get(url, stream=True).raw)
+text = ["a photo of a cat", "a photo of a dog"]
 
-- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search for image captioning. 🌎
+inputs = processor(text=text, images=image, return_tensors="pt", padding=True)
+inputs = {k: v.to("cuda") for k, v in inputs.items()}
 
-**Image retrieval**
+with torch.no_grad():
+    outputs = model(**inputs)
 
-- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing) on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
-- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb) on image retrieval and showing the similarity score. 🌎
-- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing) on how to map images and texts to the same vector space using Multilingual CLIP. 🌎 
-- A [notebook](https://colab.research.google.com/github/vivien000/clip-demo/blob/master/clip.ipynb#scrollTo=uzdFhRGqiWkR) on how to run CLIP on semantic image search using [Unsplash](https://unsplash.com) and [TMDB](https://www.themoviedb.org/) datasets. 🌎
+image_features = outputs.image_embeds
+text_features = outputs.text_embeds
 
-**Explainability**
+similarity = torch.cosine_similarity(image_features, text_features)
+```
 
-- A [notebook](https://colab.research.google.com/github/hila-chefer/Transformer-MM-Explainability/blob/main/CLIP_explainability.ipynb) on how to visualize similarity between input token and image segment. 🌎
+## Notes
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we will review it.
-The resource should ideally demonstrate something new instead of duplicating an existing resource.
+- For the best speedups, we recommend loading the model in half-precision (e.g. torch.float16 or torch.bfloat16).
```

Comment:
Not necessary at the moment since we show how to load in half-precision by default